src/util/ajax.js

```diff
@@ -158,7 +161,15 @@ function makeXMLHttpRequest(requestParameters: RequestParameters, callback: Resp
     return { cancel: () => xhr.abort() };
 }
 
-const makeRequest = window.fetch && window.Request && window.AbortController ? makeFetchRequest : makeXMLHttpRequest;
+export const makeRequest = function(requestParameters: RequestParameters, callback: ResponseCallback<any>): Cancelable {
+    if (window.fetch && window.Request && window.AbortController && !/^file:/.test(requestParameters.url)) {
+        return makeFetchRequest(requestParameters, callback);
+    } else if (isWorker() && self.worker.actor) {
+        return self.worker.actor.send('getResource', requestParameters, callback);
+    } else {
+        return makeXMLHttpRequest(requestParameters, callback);
```

Comment:
Is it intended that the worker-main hack will be used not only in IE11 context but also for file protocol requests in modern browsers, or am I misreading the conditions?